### PR TITLE
Add a tutorial on hyperparameter search for VAECF 

### PR DESCRIPTION
**Pull request by saghiles**
### Checklist:

- [ ] I have added tests.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated `README.md` (if you are adding a new model).
- [ ] I have updated `examples/README.md` (if you are adding a new example).
- [ ] I have updated `datasets/README.md` (if you are adding a new dataset).

### CODE REVIEW

File: tutorials/param_search_vaecf.ipynb [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+   "source": [
+    "# Load MovieLens 100K ratings\n",
+    "ml_100k = movielens.load_feedback(variant=\"100K\")\n",
+    "\n",
+    "# Define an evaluation method to split feedback into train, validation and test sets\n",
+    "ratio_split = RatioSplit(data=ml_100k, test_size=0.1, val_size=0.1, seed=123, verbose=False)\n",
+    "\n",
+    "# Instantiate Recall@100 and NDCG@100 for evaluation\n",
+    "rec100 = cornac.metrics.Recall(100)\n",
+    "ndcg100 = cornac.metrics.NDCG(100)\n",
```
Comment (tqtg) [inline tutorials/param_search_vaecf.ipynb]: It seems like we are not using NDCG, this can be removed.

File: tutorials/param_search_vaecf.ipynb [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# GridSearch\n",
+    "gs_vaecf = GridSearch(\n",
+    "    model=vaecf,\n",
+    "    space=[\n",
+    "        Discrete(\"beta\", [0.0, 0.2, 0.4, 0.6, 0.8, 1., 1.2, 1.4, 1.6, 1.8, 2.]),\n",
```
Comment (tqtg) [inline tutorials/param_search_vaecf.ipynb]: This can be shortened by using:

```python
np.linspace(0.0, 2.0, 11)
```

File: tutorials/param_search_vaecf.ipynb [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+    "    eval_method=ratio_split,\n",
+    "    n_trails=20,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As evident from above, there are two ways we can specify search spaces, namely `Discrete` and `Continuous`. In the latter case, `n_trails` values will be uniformly sampled in the interval [low,high). More details in the [documentation](https://cornac.readthedocs.io/en/latest/hyperopt.html)."
```
Comment (tqtg) [inline tutorials/param_search_vaecf.ipynb]: "As evident from above, there are two ways we can specify search spaces, namely `Discrete` and `Continuous`. ~~In the latter case, `n_trails` values will be uniformly sampled in the interval [low,high)~~. More details in the [documentation](https://cornac.readthedocs.io/en/latest/hyperopt.html)."

**Comment by tqtg** [comment]
I think we should mention that the best params are determined by the validation set. The validation results can be shown in addition to the test results.

**Comment by saghiles** [comment]
In the subsection run an experiment, it is mentioned that the best results are select based on the validation set. One question: are the validation results displayed by default?

**Comment by tqtg** [comment]
> In the subsection run an experiment, it is mentioned that the best results are select based on the validation set. One question: are the validation results displayed by default?

Yes, the validation results will be displayed by default if the validation set is available.

**Comment by tqtg** [comment]
@saghiles I'm good with this PR. Feel free to merge it if you have no further changes.
